File: src/main/java/de/cloudypanda/main/adventcalendar/config/AdventCalendarConfigManager.java
package de.cloudypanda.main.adventcalendar.config;

import com.fasterxml.jackson.databind.ObjectMapper;
import de.cloudypanda.main.Huntcraft;
import de.cloudypanda.main.common.file.AbstractFileManager;
import org.bukkit.Material;
import org.bukkit.enchantments.Enchantment;

import java.io.IOException;
import java.nio.file.Files;
import java.util.List;

public class AdventCalendarConfigManager extends AbstractFileManager<AdventCalendarConfigModel> {

    public AdventCalendarConfigManager(String fileName, Huntcraft huntcraft) {
        super(fileName, huntcraft, AdventCalendarConfigModel.class);
    }

    @Override
    public void afterInit() {
        setupAdventCalendar();
    }

    private void setupAdventCalendar() {
        AdventCalendarConfigModel adventCalendarConfigModel = new AdventCalendarConfigModel();

        AdventCalendarDayConfig day1Config = new AdventCalendarDayConfig();
        day1Config.setDate("2024-11-24");
        day1Config.setMessage("Welcome to the first day of the Advent Calendar! \n Today's challenge ist to submit a workbench");
        day1Config.setPoints(10);
        day1Config.setItemToSubmit(new AdventCalendarSubmitItemConfig(Material.DIAMOND_SWORD,
                1,
                null,
                null,
                List.of(new AdventCalendarSubmitItemEnchantConfig(Enchantment.SHARPNESS, 5))
        ));

        adventCalendarConfigModel.getChallenges().add(day1Config);

        this.saveToFile(adventCalendarConfigModel);
    }
}

File: src/main/java/de/cloudypanda/main/adventcalendar/config/AdventCalendarConfigManager.kt
package de.cloudypanda.main.adventcalendar.config;

import de.cloudypanda.main.Huntcraft;
import de.cloudypanda.main.common.config.AbstractFileManager;
import org.bukkit.Material;
import java.time.LocalDate

class AdventCalendarConfigManager(fileName: String, huntcraft: Huntcraft) :
    AbstractFileManager<AdventCalendarConfigModel>(fileName, huntcraft, AdventCalendarConfigModel::class.java) {

    override fun afterInit() {
        setupAdventCalendar();
    }

    private fun setupAdventCalendar() {
        val adventCalendarConfigModel = AdventCalendarConfigModel();

        val templateConfig = AdventCalendarDayConfig(
            LocalDate.now().toString(),
            "Submit a Diamond Sword",
            10,
            null,
            AdventCalendarSubmitItemConfig(
                Material.DIAMOND_SWORD,
                1,
                null,
                null,
                listOf()
            )
        );

        adventCalendarConfigModel.challenges.add(templateConfig);
        this.saveToFile(adventCalendarConfigModel);
    }
}
